src/transformers/pipelines/conversational.py

```diff
@@ -363,28 +323,23 @@ def _clean_padding_history(self, generated_tensor) -> List[List[int]]:
             outputs.append(sequence_tokens)
         return outputs
 
-    def _concat_inputs_history(self, inputs: List[List[int]], histories: List[Optional[List[int]]], max_length: int):
-        """
-        Builds an input prepended by the history for this conversation, allowing multi-turn conversation with context
-        """
-        outputs = []
-        for new_input, history in zip(inputs, histories):
-            if history is not None:
-                new_input = history + new_input
-            if len(new_input) > max_length - self.min_length_for_response:
```

Comment:
All this was actually dead-code (`new_input` is never overwritten)
Equivalent behavior (IMO) is done by

```python
  if len(input_ids) > self.tokenizer.model_max_length:
          new_input = history + new_input
      input_ids = input_ids[-self.model_max_length :]
```

This is debatable because this code actually tried to cut at EOS token boundaries, **but** a single input could be longer that this length leading to a downstream error too. Also, it's not really clear as to why cutting at EOS token boundaries is better (there's still no way for the model to know who is taking at each turn because locutor is not taken into account).

In any case, this code can't stay as is, because it's currently broken.